src/Sarif/FileRegionsCache.cs

```diff
@@ -156,11 +159,10 @@ public Region ConstructMultilineContextSnippet(Region inputRegion, Uri uri, stri
                 return null;
             }
 
-            const int bigSnippetLength = 512;
-            const int smallSnippetLength = 128;
+            fileText ??= newLineIndex.Text;
 
             // Generating full inputRegion to prevent issues.
-            Region originalRegion = this.PopulateTextRegionProperties(inputRegion, uri, populateSnippet: true);
+            Region originalRegion = this.PopulateTextRegionProperties(inputRegion, uri, populateSnippet: true, fileText);
```

Comment:
>fileText [](http://example.com/codeflow?start=111&length=8)

We were not consistent around passing this optional argument around. This likely would break certain in-memory analysis operations that aren't highly exercised/well-tested today.